firebase-sessions/src/main/kotlin/com/google/firebase/sessions/SessionLifecycleService.kt

```diff
@@ -28,14 +28,21 @@ import android.os.Looper
 import android.os.Message
 import android.os.Messenger
 import android.util.Log
+import com.google.firebase.Firebase
+import com.google.firebase.app
 import java.util.concurrent.LinkedBlockingQueue
+import java.util.concurrent.atomic.AtomicReference
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
 
 /**
  * Service for monitoring application lifecycle events and determining when/if a new session should
  * be generated. When this happens, the service will broadcast the updated session id to all
  * connected clients.
  */
-internal class SessionLifecycleService : Service() {
+internal class SessionLifecycleService(
+  private var datastore: SessionDatastore = SessionDatastore(Firebase.app.applicationContext))
```

Comment:
Is Firebase.app guaranteed to exist when this is constructed?

Reply:
I think so. Isn't it created first?